src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -326,6 +326,20 @@ public function testGetMimeTypeFromFormat($format, $mimeTypes)
         }
     }
 
+    /**
+     * @dataProvider getFormatToMimeTypeMapProviderForAccept
+     */
+    public function testGetMimeTypeFromFormatWithAccept($format, $mimeTypes, $accepts)
+    {
+        if (null !== $format) {
+            $request = new Request();
+            $request->headers->set('Accept', $accepts);
+            $request->setFormat($format, $mimeTypes);
+            //Only when accept value matches a value verbatim inside of the mimetype array.
+            $this->assertEquals($accepts, $request->getMimeType($format));
```

Comment:
This test looks wrong. `accept` is always returned by `getMimeType` and the `$format` is never tested.

My point is that the following test fails (and this is a BC break):

```php
<?php

$request = new Request();
$request->headers->set('Accept', 'application/javascript');

$this->assertEquals('text/plain', $request->getMimeType('txt'));
```